src/builder/QuarkBuilder.sol

```diff
@@ -52,6 +52,145 @@ contract QuarkBuilder {
     /* ===== Helper Functions ===== */
 
     /* ===== Main Implementation ===== */
+    struct CometRepayIntent {
+        uint256 amount;
+        string assetSymbol;
+        uint256 blockTimestamp;
+        uint256 chainId;
+        uint256[] collateralAmounts;
+        string[] collateralAssetSymbols;
+        address comet;
+        address repayer;
+    }
+
+    function cometRepay(
+        CometRepayIntent memory repayIntent,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        PaymentInfo.Payment memory payment
+    ) external pure returns (BuilderResult memory /* builderResult */ ) {
+        if (repayIntent.collateralAmounts.length != repayIntent.collateralAssetSymbols.length) {
+            revert InvalidInput();
+        }
+
+        // XXX confirm that the user is not withdrawing beyond their limits
+
+        assertFundsAvailable(
+            repayIntent.chainId, repayIntent.assetSymbol, repayIntent.amount, chainAccountsList, payment
+        );
+
+        uint256 actionIndex = 0;
+        Actions.Action[] memory actions = new Actions.Action[](chainAccountsList.length);
+        IQuarkWallet.QuarkOperation[] memory quarkOperations =
+            new IQuarkWallet.QuarkOperation[](chainAccountsList.length);
```

Comment:
Since the dynamic list is checked in, probably will be good to use dynamic list instead of tracking actionIndex in places.

Reply:
will make that update in a follow-up PR; I want to get Borrow and Repay merged to unblock end-to-end testing